### PR TITLE
Abstract edge case fixes

### PR DESCRIPTION
**Pull request by MacondoExpress**
Support concrete entitiy filtering and interface shared filters 
```
Tests improved in branch B:  [
  'Interface Relationships Interface Relationship Query through connection with where excluding type',
  'Interface Relationships Interface Relationship Query through connection with where overriding',
  'Interface Relationships Simple Interface Relationship Query with offset and limit',
  'Interface Relationships Type filtering using onType'
]
```

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3957.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

File: packages/graphql/src/translate/queryAST/ast/operations/interfaces/InterfaceReadOperation.ts [hunk 1]
```diff
@@ -103,4 +107,8 @@ export class InterfaceReadOperation extends Operation {
     public addSort(...sortElement: Sort[]): void {
         this.sortFields.push(...sortElement);
     }
+
+    protected getSortFields(context: QueryASTContext, target: Cypher.Variable): SortField[] {
+        return this.sortFields.flatMap((sf) => sf.getSortFields(context, target, false));
+    }
```
Comment (angrykoala) [inline packages/graphql/src/translate/queryAST/ast/operations/interfaces/InterfaceReadOperation.ts:113]: I think the name of this function this may lead to a problem in the future

`getSortFields` is the name used in the contract for sort, which is "public" in class `Sort`

In this case, we have a protected function with the same name. At the moment is not a problem, but in the future we will probably try to merge all the interfaces to have a common contract between all nodes

I think we should consider renaming this protected function, to avoid confusion/conflicts with the contract in the future

File: packages/graphql/src/translate/queryAST/factory/OperationFactory.ts [hunk 2]
```diff
@@ -241,25 +217,52 @@ export class OperationsFactory {
...
 
                     return this.hydrateConnectionOperationAST({
                         relationship,
                         target: concreteEntity,
                         resolveTree,
                         context,
                         operation: connectionPartial,
+                        whereArgs: whereArgs,
                     });
-                }
```
Comment (angrykoala) [inline packages/graphql/src/translate/queryAST/factory/OperationFactory.ts:261]: These big blocks are getting complex to follow, please move them to separate functions if possible

File: packages/graphql/src/translate/queryAST/factory/OperationFactory.ts [hunk 3]
```diff
@@ -320,14 +323,16 @@ export class OperationsFactory {
...
     }: {
         relationship: RelationshipAdapter;
         target: ConcreteEntityAdapter;
         resolveTree: ResolveTree;
         context: Neo4jGraphQLTranslationContext;
         operation: T;
+        whereArgs: Record<string, any>;
     }): T {
-        let whereArgs = (resolveTree.args.where || {}) as Record<string, any>;
+        //let whereArgs = (resolveTree.args.where || {}) as Record<string, any>;
```
Comment (angrykoala) [inline packages/graphql/src/translate/queryAST/factory/OperationFactory.ts:335]: :eyes:

File: packages/graphql/src/translate/queryAST/utils/filter-concrete-entities-with-on-filter.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import type { ConcreteEntityAdapter } from "../../../schema-model/entity/model-adapters/ConcreteEntityAdapter";
+import type { InterfaceEntityAdapter } from "../../../schema-model/entity/model-adapters/InterfaceEntityAdapter";
+import type { UnionEntityAdapter } from "../../../schema-model/entity/model-adapters/UnionEntityAdapter";
+import { isUnionEntity } from "./is-union-entity";
+
+/**
+ * Returns the concrete entities presents in the where [_on] argument,
+ * if the where argument is not defined then returns all the concrete entities of the composite target.
+ **/
+export function filterConcreteEntities(
```
Comment (angrykoala) [inline packages/graphql/src/translate/queryAST/utils/filter-concrete-entities-with-on-filter.ts:10]: This name does not match the file name.

Could we try a more explicit name?

```
getConcreteEntitiesInOnArgumentOfWhere
```

That's a bit verbose, so maybe there is a better name

And update the file so the name is the same

File: packages/graphql/src/translate/queryAST/ast/operations/interfaces/InterfaceReadOperation.ts (hunk 1 above)
Comment (MacondoExpress) [reply packages/graphql/src/translate/queryAST/ast/operations/interfaces/InterfaceReadOperation.ts:113]: Sure, I agree! Just a thing, this name and behavior mimic what we have in the class InterfaceConnectionReadOperation, I tried to keep everything as close as possible between these classes so that will be easier to unify later!

File: packages/graphql/src/translate/queryAST/utils/filter-concrete-entities-with-on-filter.ts (hunk 4 above)
Comment (MacondoExpress) [reply packages/graphql/src/translate/queryAST/utils/filter-concrete-entities-with-on-filter.ts:10]: Good catch, I've played around with it and I've losed track of the name of it!

File: packages/graphql/src/translate/queryAST/ast/operations/interfaces/InterfaceReadOperation.ts (hunk 1 above)
Comment (angrykoala) [reply packages/graphql/src/translate/queryAST/ast/operations/interfaces/InterfaceReadOperation.ts:113]: Oh, that's great, is just so we can distinguish that this is not yet the same interface :)
